pkg/testing/runner/runner.go

```diff
@@ -0,0 +1,858 @@
+	if ps.ExitCode() != 0 {
+		return fmt.Errorf("failed to run ogc import: docker run exited with code: %d", ps.ExitCode())
+	}
+	return nil
+}
+
+// ogcUp brings up all the instances.
+func (r *Runner) ogcUp(ctx context.Context) ([]byte, error) {
+	fmt.Printf(">>> Bring up instances through ogc\n")
+	var output bytes.Buffer
+	proc, err := r.ogcRun(ctx, []string{"up", LayoutIntegrationTag}, false, process.WithCmdOptions(attachOut(&output), attachErr(&output)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ogc up: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		// print the output so its clear what went wrong
+		fmt.Printf("%s\n", output.Bytes())
+		return nil, fmt.Errorf("failed to run ogc up: docker run exited with code: %d", ps.ExitCode())
+	}
+	return output.Bytes(), nil
+}
+
+// ogcDown brings down all the instances.
+func (r *Runner) ogcDown(ctx context.Context) error {
+	fmt.Printf(">>> Bring down instances through ogc\n")
+	proc, err := r.ogcRun(ctx, []string{"down", LayoutIntegrationTag}, false)
+	if err != nil {
+		return fmt.Errorf("failed to run ogc down: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		return fmt.Errorf("failed to run ogc down: docker run exited with code: %d", ps.ExitCode())
+	}
+	return nil
+}
+
+// ogcMachines brings up all the instances.
+func (r *Runner) ogcMachines(ctx context.Context) ([]OGCMachine, error) {
+	var out bytes.Buffer
+	proc, err := r.ogcRun(ctx, []string{"ls", "--as-yaml"}, false, process.WithCmdOptions(attachOut(&out)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ogc ls: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		return nil, fmt.Errorf("failed to run ogc ls: docker run exited with code: %d", ps.ExitCode())
+	}
+	var machines []OGCMachine
+	err = yaml.Unmarshal(out.Bytes(), &machines)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse ogc ls output: %w", err)
+	}
+	return machines, nil
+}
+
+func (r *Runner) ogcRun(ctx context.Context, args []string, interactive bool, processOpts ...process.StartOption) (*process.Info, error) {
+	wd, err := r.getWorkDir()
+	if err != nil {
+		return nil, err
+	}
+	tokenName := filepath.Base(r.cfg.GCE.ServiceTokenPath)
+	clientEmail, err := r.cfg.GCE.ClientEmail()
+	if err != nil {
+		return nil, err
+	}
+	projectID, err := r.cfg.GCE.ProjectID()
+	if err != nil {
+		return nil, err
+	}
+	runArgs := []string{"run"}
+	if interactive {
+		runArgs = append(runArgs, "-i")
+	}
+	runArgs = append(runArgs,
+		"--rm",
+		"-e",
+		fmt.Sprintf("GOOGLE_APPLICATION_SERVICE_ACCOUNT=%s", clientEmail),
+		"-e",
+		fmt.Sprintf("GOOGLE_APPLICATION_CREDENTIALS=/root/%s", tokenName),
+		"-e",
+		fmt.Sprintf("GOOGLE_PROJECT=%s", projectID),
+		"-e",
+		fmt.Sprintf("GOOGLE_DATACENTER=%s", r.cfg.GCE.Datacenter),
+		"-v",
+		fmt.Sprintf("%s:/root/%s", r.cfg.GCE.ServiceTokenPath, tokenName),
+		"-v",
+		fmt.Sprintf("%s:%s", wd, wd),
+		"-w",
+		wd,
+		"docker.io/gorambo/ogc:blake",
```

Comment:
Is this a stable tag ? :thinking:

Reply:
It is at the moment, I will do a follow up PR, once it is set to `:latest`.

Reply:
I have updated this to `:latest`, as its now the most recent changes are merged.